controllers/quiz.js

```diff
@@ -0,0 +1,136 @@
+
+const { throwError } = require("../utils/helper");
+
+const quizDetails=require('../models/quizDetails')
+const questions=require('../models/questions')
+
+
+
+exports.createQuiz=(async (req,res,next)=>{
+    const edit=req.query.edit;
+    const quiz_name=req.body.quiz_name
+    if(edit && edit.toLowerCase() == 'true'){
+        const id=req.body.quiz_id
+        try{
+          const result=await quizDetails.updateQuizName(id,quiz_name);
+            if(result.rowsAffected.length > 0){
+
+          return res.status(200).json({msg: "Details of Resource Person Successfully updated"});
+            }
+            else{
+                throwError("Something went Wrong",400);
+            }
+        }
+        catch(err){
+          return next(err)
+        }
+    }
+    else{
+    try{
+    const quiz=new quizDetails(quiz_name);
+    await quiz.addQuizDetails();
+    return res.status(201).json({msg: "New Quiz Created"});
```

Comment:
**Quiz_id** should also be added in **workshop_details** table. Otherwise we won't know which quiz is associated with which workshop.